webrob/neems/manager.py

```diff
@@ -1,23 +1,39 @@
 import os
 from flask import session
 from webrob.neems.neem import NEEM
-from webrob.app_and_db import app, mongoDBMetaCollection
+from webrob.app_and_db import app, get_mongo_db_meta_collection
+from pymongo.errors import ConnectionFailure, PyMongoError
+from webrob.models.NEEMMetaException import NEEMMetaException
+from webrob.models.NEEMHubSettings import get_settings_count
 
 NEEM_DIR = "/neems"
 
 
 class NEEM_Manager:
     def __init__(self):
-        self.neem_ids = self.__get_neem_ids__()
-        # TODO: remove again, just for testing pagination as long as only few neems
-        #        are there
-        self.neem_ids = self.neem_ids * 20
+        self.neem_ids = self.set_neem_ids()
+
+    def set_neem_ids(self):
+        self.neem_ids = []
 
-    def __get_neem_ids__(self):
-        neem_ids = []
         # get all neem ids information from collection
-        neem_ids = mongoDBMetaCollection.find().distinct('_id')
-        return neem_ids
+        # first find if there is any existing entry in db
+        count = get_settings_count()
```

Comment:
when application starts for the first time, it goes through manager and then subsequently in get_mongo_db_meta_collection but the exception thrown there results into an error. I am not 100% sure why this happens. Thus in order to avoid when db is either empty or not initialized, I would skip calling get_mongo_db_meta_collection.